lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -0,0 +1,156 @@
+  #clearTimeout = FunctionPrototypeBind(this.#clearTimer, this);
+  #setInterval = FunctionPrototypeBind(this.#createTimer, this, true);
+  #clearInterval = FunctionPrototypeBind(this.#clearTimer, this);
+
+  constructor() {
+    emitExperimentalWarning('The MockTimers API');
+  }
+
+  #createTimer(isInterval, callback, delay, ...args) {
+    const timerId = this.#currentTimer++;
+    this.#timers.insert({
+      __proto__: null,
+      id: timerId,
+      callback,
+      runAt: DateNow() + delay,
+      interval: isInterval,
+      args,
+    });
+
+    return timerId;
+  }
+
+  #clearTimer(position) {
+    this.#timers.removeAt(position);
+  }
+
+  tick(time = 1) {
+
+    // if (!this.isEnabled) {
+    //   throw new Error('you should enable fakeTimers first by calling the .enable function');
+    // }
+    this.#now += time;
+
+    for (let timer = this.#timers.peek(); timer && timer.runAt <= this.#now; timer = this.#timers.peek()) {
+      timer.callback(...timer.args);
+      this.#timers.shift();
+      if (timer.interval) {
+        timer.runAt += timer.interval;
+        this.#timers.insert(timer);
+      }
+    }
+
+  }
+
+  enable() {
+    // if (this.isEnabled) {
+    //   throw new Error('fakeTimers is already enabled!');
+    // }
+    this.#now = DateNow();
+    this.#isEnabled = true;
+
+    this.#realSetTimeout = globalThis.setTimeout;
+    this.#realClearTimeout = globalThis.clearTimeout;
+    this.#realSetInterval = globalThis.setInterval;
+    this.#realClearInterval = globalThis.clearInterval;
+
+    this.#realTimersSetTimeout = nodeTimers.setTimeout;
+    this.#realTimersClearTimeout = nodeTimers.clearTimeout;
+    this.#realTimersSetInterval = nodeTimers.setInterval;
+    this.#realTimersClearInterval = nodeTimers.clearInterval;
+
+    this.#realPromisifiedSetTimeout = nodeTimersPromises.setTimeout;
+
+    globalThis.setTimeout = this.#setTimeout;
+    globalThis.clearTimeout = this.#clearTimeout;
+    globalThis.setInterval = this.#setInterval;
+    globalThis.clearInterval = this.#clearInterval;
+
+    nodeTimers.setTimeout = this.#setTimeout;
+    nodeTimers.clearTimeout = this.#clearTimeout;
+    nodeTimers.setInterval = this.#setInterval;
+    nodeTimers.clearInterval = this.#clearInterval;
+
+    nodeTimersPromises.setTimeout = (ms) => new Promise(
+      (resolve) => {
+        const id = this.#setTimeout(() => resolve(id), ms);
+      },
+    );
+
+  }
+
+  reset() {
+    this.#isEnabled = false;
+
+    // Restore the real timer functions
+    globalThis.setTimeout = this.#realSetTimeout;
+    globalThis.clearTimeout = this.#realClearTimeout;
+    globalThis.setInterval = this.#realSetInterval;
+    globalThis.clearInterval = this.#realClearInterval;
+
+    nodeTimers.setTimeout = this.#realTimersSetTimeout;
+    nodeTimers.clearTimeout = this.#realTimersClearTimeout;
+    nodeTimers.setInterval = this.#realTimersSetInterval;
+    nodeTimers.clearInterval = this.#realTimersClearInterval;
```

Comment:
it might be easier to maintain to have a private method that both "enable" and "reset" call - that way the mutations can all be done in one place, and it'll be harder to have the two places go out of sync

Reply:
nice! will do that! I'm gonna refactor more those functions. Gonna focus today on delivering most of the code I tomorrow start refactoring it 🤩